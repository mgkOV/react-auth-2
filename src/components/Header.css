.logo-icon {
  /*color: #ab47bc;*/
  margin-left: 20px;
  font-size: 3.8rem;
}
