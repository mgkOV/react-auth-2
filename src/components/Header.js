import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink, withRouter } from 'react-router-dom';
import { logOut } from '../actions';
import MdFitnessCenter from 'react-icons/lib/md/fitness-center';

import './Header.css';

export class Header extends Component {
  handleLogOut = e => {
    e.preventDefault();
    this.props.logOut();
    this.props.history.push('/signin');
  };

  render() {
    const { auth: { authenticated } } = this.props;
    return (
      <nav>
        <div className="nav-wrapper indigo darken-1">
          <NavLink to="/" className="brand-logo">
            <MdFitnessCenter className="logo-icon" />
          </NavLink>
          <ul id="nav-mobile" className="right hide-on-med-and-down">
            <li>
              {authenticated
                ? <a onClick={this.handleLogOut}>Выйти</a>
                : <NavLink to="/signin">Войти</NavLink>}
            </li>
          </ul>
        </div>
      </nav>
    );
  }
}

export default connect(
  ({ auth }) => ({
    auth
  }),
  { logOut }
)(withRouter(Header));
