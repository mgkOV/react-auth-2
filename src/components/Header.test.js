import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header', () => {
  const header = shallow(<Header />);

  it('should render properly', () => {
    expect(header).toMatchSnapshot();
  });
});
