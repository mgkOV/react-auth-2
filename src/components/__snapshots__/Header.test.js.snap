// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="nav-wrapper indigo darken-1"
>
        <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="brand-logo"
                to="/"
        >
                <MdFitnessCenter
                        className="logo-icon"
                />
        </NavLink>
        <ul
                className="right hide-on-med-and-down"
                id="nav-mobile"
        >
                <li>
                        <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/signin"
                        >
                                Войти
                        </NavLink>
                </li>
        </ul>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="brand-logo"
            to="/"
>
            <MdFitnessCenter
                        className="logo-icon"
            />
</NavLink>,
          <ul
            className="right hide-on-med-and-down"
            id="nav-mobile"
>
            <li>
                        <NavLink
                                    activeClassName="active"
                                    ariaCurrent="true"
                                    to="/signin"
                        >
                                    Войти
                        </NavLink>
            </li>
</ul>,
        ],
        "className": "nav-wrapper indigo darken-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <MdFitnessCenter
              className="logo-icon"
/>,
            "className": "brand-logo",
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "logo-icon",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <li>
              <NavLink
                            activeClassName="active"
                            ariaCurrent="true"
                            to="/signin"
              >
                            Войти
              </NavLink>
</li>,
            "className": "right hide-on-med-and-down",
            "id": "nav-mobile",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/signin"
>
                Войти
</NavLink>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Войти",
                "to": "/signin",
              },
              "ref": null,
              "rendered": "Войти",
              "type": [Function],
            },
            "type": "li",
          },
          "type": "ul",
        },
      ],
      "type": "div",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="nav-wrapper indigo darken-1"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="brand-logo"
                    to="/"
          >
                    <MdFitnessCenter
                              className="logo-icon"
                    />
          </NavLink>
          <ul
                    className="right hide-on-med-and-down"
                    id="nav-mobile"
          >
                    <li>
                              <NavLink
                                        activeClassName="active"
                                        ariaCurrent="true"
                                        to="/signin"
                              >
                                        Войти
                              </NavLink>
                    </li>
          </ul>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="brand-logo"
              to="/"
>
              <MdFitnessCenter
                            className="logo-icon"
              />
</NavLink>,
            <ul
              className="right hide-on-med-and-down"
              id="nav-mobile"
>
              <li>
                            <NavLink
                                          activeClassName="active"
                                          ariaCurrent="true"
                                          to="/signin"
                            >
                                          Войти
                            </NavLink>
              </li>
</ul>,
          ],
          "className": "nav-wrapper indigo darken-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": <MdFitnessCenter
                className="logo-icon"
/>,
              "className": "brand-logo",
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "logo-icon",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <li>
                <NavLink
                                activeClassName="active"
                                ariaCurrent="true"
                                to="/signin"
                >
                                Войти
                </NavLink>
</li>,
              "className": "right hide-on-med-and-down",
              "id": "nav-mobile",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  ariaCurrent="true"
                  to="/signin"
>
                  Войти
</NavLink>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "ariaCurrent": "true",
                  "children": "Войти",
                  "to": "/signin",
                },
                "ref": null,
                "rendered": "Войти",
                "type": [Function],
              },
              "type": "li",
            },
            "type": "ul",
          },
        ],
        "type": "div",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
